lib/rspec/core.rb

```diff
@@ -31,6 +31,7 @@
   option_parser
   configuration_options
   runner
+  profiler
```

Comment:
> but in order to make `config#profile_examples` dynamic we must always have the profiler on

A big reason tracking example group execution time in a registered formatter (even if you don't call it one but I do!) appealed to me is that we want the added runtime cost of tracking times for profiling to only be paid if the user has opted in via `--profile`.  It makes me a sad panda to always pay the cost of tracking that when the user hasn't turned on profiling.

IMO, we don't need to support dynamic changes to `config#profile_examples` after the suite has started running.  We can treat it like adding a formatter: it has to be done before the suite starts to take affect.  (A user obviously can't expect to turn on profiling half way through the suite run and expect it to work).

My problem with what you had before is that `config` exposed the `profile_examples` option and a derived attribute, and it seems bad for an object to expose a derived attribute but then not keep it in sync with the source value.  Now that you're not exposing the profiler object off of `config`, I'm not concerned about that anymore.  Now I just view `profile_examples` as a boolean toggle that is checked once, when setting things up to start running the suite, and ignored the rest of the time.  Users can toggle it back and forth all they want before the suite starts and it'll work fine still.

In effect, I think we should only load the profiler and register it with the reporter if we get [here](https://github.com/rspec/rspec-core/blob/5375dcdfdb5460a51d5003de76eb124997fa26ea/lib/rspec/core/formatters.rb#L126).  That'll avoid the cost of loading and using it if the user hasn't opted-in to it.